.circleci/config.yml

```diff
@@ -10,16 +10,12 @@ commands:
       subcommand:
         type: string
     steps:
-      - run:
-          name: Install dependencies
-          command: |
-            ln -fs /usr/share/zoneinfo/America/New_York /etc/localtime
-            apt update
-            apt install -y build-essential zlib1g-dev uuid-dev libdigest-sha-perl libelf-dev bc bzip2 bison flex git gnupg gawk iasl m4 nasm patch python python2 python3 wget gnat cpio ccache pkg-config cmake libusb-1.0-0-dev autoconf texinfo ncurses-dev doxygen graphviz udev libudev1 libudev-dev automake libtool rsync innoextract sudo libssl-dev device-tree-compiler u-boot-tools sharutils e2fsprogs parted curl unzip imagemagick libncurses5-dev zip
       - run:
           name: Make Board (FULL ORDERED BUILD LOGS HERE UNTIL JOB FAILED)
           command: |
-            rm -rf build/<<parameters.arch>>/<<parameters.target>>/* build/<<parameters.arch>>/log/* && make V=1 BOARD=<<parameters.target>> <<parameters.subcommand>> || touch ./tmpDir/failed_build
+            rm -rf build/<<parameters.arch>>/<<parameters.target>>/* build/<<parameters.arch>>/log/* &&
+            nix --extra-experimental-features 'nix-command flakes' develop --ignore-environment --command make V=1 BOARD=<<parameters.target>> <<parameters.subcommand>> ||
```

Comment:
Do we have any idea how reliable is the "experimental feature" here going to be in the long term?  Particularly given since the goal here is to enable a reproducible build environment, if the flakes experimental feature changes incompatibly it will undermine that goal.  Is there any way to do this without using experimental features?

Reply:
https://nixos.org/manual/nix/stable/contributing/experimental-features.html

@mmlb I could not easily find if flakes are there to stay. Can you feed us in that subject?

Reply:
nix flakes are very much here to stay imo. There's active work to get flakes and the unified cli out from the experimental label upstream. https://determinate.systems/posts/experimental-does-not-mean-unstable is a nice overview imo and echoes my experience.

Reply:
Let's say it's not that simple. Here's the most recent attempt at a sober summary of the state of affairs: https://nix.dev/concepts/flakes

In any case you should layout your Nix expressions such that they can be used regardless of flakes or not, for instance by always providing a proper `default.nix` (and possibly `shell.nix`) as an entry point for stable Nix, and exposing recipes and libraries such that they can be used with `callPackage`.